pkg/services/ngalert/ngalert.go

```diff
@@ -0,0 +1,96 @@
+package ngalert
+
+import (
+	"time"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// AlertNG is the service for evaluating the condition of an alert definition.
+type AlertNG struct {
+	Bus             bus.Bus                  `inject:""`
+	Cfg             *setting.Cfg             `inject:""`
+	DatasourceCache datasources.CacheService `inject:""`
+	RouteRegister   routing.RouteRegister    `inject:""`
+	SQLStore        *sqlstore.SqlStore       `inject:""`
+	log             log.Logger
+}
+
+func init() {
+	registry.RegisterService(&AlertNG{})
+}
+
+// Init initializes the AlertingService.
+func (ng *AlertNG) Init() error {
+	ng.log = log.New("ngalert")
+
+	ng.registerAPIEndpoints()
+
+	return nil
+}
+
+// IsDisabled returns true if the alerting service is disable for this instance.
+func (ng *AlertNG) IsDisabled() bool {
+	if ng.Cfg == nil {
+		return false
+	}
+	// Check also about expressions?
+	return !ng.Cfg.IsNgAlertEnabled()
+}
+
+// AddMigration defines database migrations.
+// If Alerting NG is not enabled does nothing.
+func (ng *AlertNG) AddMigration(mg *migrator.Migrator) {
+	if ng.IsDisabled() {
+		return
+	}
+
+	alertDefinition := migrator.Table{
+		Name: "alert_definition",
+		Columns: []*migrator.Column{
+			{Name: "id", Type: migrator.DB_BigInt, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "org_id", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "name", Type: migrator.DB_NVarchar, Length: 255, Nullable: false},
+			{Name: "condition", Type: migrator.DB_NVarchar, Length: 255, Nullable: false},
+			{Name: "data", Type: migrator.DB_Text, Nullable: false},
+		},
+		Indices: []*migrator.Index{
+			{Cols: []string{"org_id"}, Type: migrator.IndexType},
+		},
+	}
+	// create table
+	mg.AddMigration("create alert_definition table", migrator.NewAddTableMigration(alertDefinition))
+
+	// create indices
+	mg.AddMigration("add index alert_definition org_id", migrator.NewAddIndexMigration(alertDefinition, alertDefinition.Indices[0]))
+}
+
+// LoadAlertCondition returns a Condition object for the given alertDefinitionID.
+func (ng *AlertNG) LoadAlertCondition(alertDefinitionID int64, signedInUser *models.SignedInUser, skipCache bool) (*eval.Condition, error) {
```

Comment:
This doesn't need to be public:
```suggestion
// loadAlertCondition returns a Condition object for the given alertDefinitionID.
func (ng *AlertNG) loadAlertCondition(alertDefinitionID int64, signedInUser *models.SignedInUser, skipCache bool) (*eval.Condition, error) {
```